### PR TITLE
Add feature to reconnect WebSocket from client to server

### PR DESCRIPTION
**Pull request by ac-61**
This feature allows the WebSocket on the client side to reconnect to the server if it is ever disconnected. The client checks if the WebSocket connection is open every minute and if the connection is closed, it will attempt to retry to connect. If the reconnection fails, the interval between reconnection attempts doubles for every failure.

A toast is displayed both when connected and disconnected from server WebSocket.

### CODE REVIEW

**Review by tariqksoliman — CHANGES_REQUESTED** [review]
Main functionality looks perfect!

Could you get rid of the first connection toast since it's kind of annoying. Only show Successfully Connected if it had previously failed. Also shift the toasts to be only ~5 pixels from the right side.

**Comment by ac-61** [comment]
@tariqksoliman I made the requested changes!

File: src/essence/essence.js [hunk 1]
```diff
@@ -125,6 +126,147 @@ $(document.body).keydown(function (e) {
...
+            console.log('Websocket connection opened...')
+
+            UserInterface_.removeLayerUpdateButton()
+
+            M.Toast.dismissAll()
+
+            if (essence.webSocketRetryInterval > essence.initialWebSocketRetryInterval) {
+                M.toast({
+                    html: "Successfully connected to WebSocket",
+                    displayLength: 160000000,
```
Comment (tariqksoliman) [inline src/essence/essence.js:174]: Nice! And this is a lot of milliseconds.

Comment (ac-61) [reply src/essence/essence.js]: Crud, I would say something about Monday but it's Tuesday now! Fixed with 3b4cdd011e0296db05cdc2d80ba92e358eeb2164!

**Review by tariqksoliman — APPROVED** [review]
LGTM!

**Pull request merged**